src/components/cards/FiatAmountInputCard.tsx

```diff
@@ -41,29 +41,31 @@ const FiatAmountInputCardComponent = ({ wallet, iconUri, inputModalMessage, titl
     const token = tokenId != null ? allTokens[tokenId] : null
     const { denominations: destDenoms } = token != null ? token : wallet.currencyInfo
     const destExchangeMultiplier = destDenoms == null ? '0' : destDenoms[0].multiplier
-
+    
+    // Clean localized fiat amount prior to biggystring ops
+    const fiatAmount = sanitizeDecimalAmount(inputFiatAmount, 2)
     const calculatedNativeCryptoAmount = truncateDecimals(
       destToFiatRate == null || destToFiatRate === '0' ? '0' : mul(destExchangeMultiplier, div(fiatAmount ?? '0', destToFiatRate, DECIMAL_PRECISION)),
       0
     )
 
     setNativeCryptoAmount(calculatedNativeCryptoAmount)
-    if (!zeroString(calculatedNativeCryptoAmount) && !zeroString(fiatAmount)) onAmountChanged(fiatAmount, calculatedNativeCryptoAmount)
-  }, [allTokens, destToFiatRate, fiatAmount, onAmountChanged, tokenId, wallet.currencyInfo])
+    if (!zeroString(calculatedNativeCryptoAmount) && !zeroString(fiatAmount)) onAmountChanged(inputFiatAmount, calculatedNativeCryptoAmount)
+  }, [allTokens, destToFiatRate, inputFiatAmount, onAmountChanged, tokenId, wallet.currencyInfo])
 
   const handleEditActionfiatAmount = React.useCallback(() => {
     Airship.show<string | undefined>(bridge => <TextInputModal title={title} message={inputModalMessage} bridge={bridge} keyboardType="decimal-pad" />).then(
       inputAmount => {
-        if (inputAmount != null && !zeroString(inputAmount)) {
-          setFiatAmount(inputAmount)
-          onAmountChanged(fiatAmount, nativeCryptoAmount)
+        if (inputAmount != null && !zeroString(sanitizeDecimalAmount(inputAmount, 2))) {
+          setInputFiatAmount(inputAmount)
+          onAmountChanged(inputFiatAmount, nativeCryptoAmount)
```

Comment:
Passing an unsantized user input to onAmountChanged isn't safe. This value is passed to a biggystring div function in LoanCreateScene so the bug still exists. 

You can remove a lot of risk for bugs like this by sanitizing immediately upon user input before doing anything else. Take the return value from Airship and sanitize it before entering this if block. Then every usage of it zeroString, onAmountChanged, and beyond, are using 
a safe and nice decimal amount. If the value needs to be displayed to the user you can convert it back where it gets rendered using utils from locales.intl.ts